src/Query/Mysql/BinToUuid.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace DoctrineExtensions\Query\Mysql;
+
+use Doctrine\ORM\Query\AST\Functions\FunctionNode;
+use Doctrine\ORM\Query\Lexer;
+use Doctrine\ORM\Query\Parser;
+use Doctrine\ORM\Query\SqlWalker;
+
+class BinToUuid extends FunctionNode
+{
+    public $binaryUuid = null;
+
+    public $swapFlag = null;
+
+    public function parse(Parser $parser): void
+    {
+        $parser->match(Lexer::T_IDENTIFIER);
+        $parser->match(Lexer::T_OPEN_PARENTHESIS);
+
+        $this->binaryUuid = $parser->ArithmeticPrimary();
+
+        if (Lexer::T_COMMA === $parser->getLexer()->lookahead['type']) {
```

Comment:
I'm not sure if should use `lookahead['type']` or `lookahead->type` here. Let me know.

Reply:
Saw in #40 that should be the property way, so replaced it.